cmd/geth/snapshot.go

```diff
@@ -149,6 +166,55 @@ It's also usable without snapshot enabled.
 	}
 )
 
+func pruneBlock(ctx *cli.Context) error {
```

Comment:
We need some protection here. Make sure the MPT and snapshot matches before when doing prune, otherwise, the node can not start.

Reply:
Using the check like this?
`snapshot.New(db, trie.NewDatabase(db), 256, int(triesInMemory), headBlock.Root(), false, false, false)`